scc/light_client/provider/rpc.go

```diff
@@ -0,0 +1,118 @@
+package provider
+
+import (
+	"fmt"
+
+	"github.com/0xsoniclabs/sonic/ethapi"
+	"github.com/0xsoniclabs/sonic/scc"
+	"github.com/0xsoniclabs/sonic/scc/cert"
+	"github.com/Fantom-foundation/lachesis-base/inter/idx"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// RPCProvider implements the Provider interface and provides methods
+// making RPC calls through an Ethereum client.
+type RPCProvider struct {
+	// client is the Ethereum client used for making RPC calls.
+	client *ethclient.Client
+}
+
+// NewRPCProviderFromClient creates a new instance of RPCProvider with the given
+// Ethereum client.
```

Comment:
nit: you should mention that the new provider instance takes ownership of the provided client and will close it if the resulting provider is closed.